core/types/data_source_external.go

```diff
@@ -33,15 +37,34 @@ func (e *DataSourceDefinitionExternal) IntoProto() *vegapb.DataSourceDefinitionE
 
 func (e *DataSourceDefinitionExternal) String() string {
 	if e.SourceType != nil {
-		return e.SourceType.String()
+		switch dsn := e.SourceType.oneOfProto().(type) {
+		case *vegapb.DataSourceDefinitionExternal_Oracle:
+			if dsn.Oracle != nil {
+				return dsn.Oracle.String()
+			}
+		case *vegapb.DataSourceDefinitionExternal_EthOracle:
+			if dsn.EthOracle != nil {
+				return dsn.EthOracle.String()
+			}
+		}
 	}
 
 	return ""
 }
 
 func (e *DataSourceDefinitionExternal) DeepClone() dataSourceType {
```

Comment:
BTW why do we do all this deep cloning, aren't these specs effectively immutable?

Reply:
Removed them now, we do not need them actually, thanks!